Gems/ONNX/README.md

```diff
@@ -38,3 +38,139 @@ This is an experimental gem implementing [ONNX Runtime](https://onnxruntime.ai/)
+- It has also recently become a member of the [Linux Foundation](https://www.linuxfoundation.org/)
+
+### **Why do we need an ONNX Gem in O3DE?**
+- We want to give users the ability to easily inference ONNX models within the engine and measure inference performance
+- There are many uses for ML within games, such as Terrain Generation, ML Agents, Computer Vision and Animation
+- The existing [ONNX Runtime C++ API](https://onnxruntime.ai/docs/api/c/namespace_ort.html) is not user friendly and requires a lot of additional steps for integration with the engine
+- Provide a plug-and-play experience, removing as much prerequisite knowledge required as possible, so that people with no ML knowledge can still implement ML model inferencing using ONNX Runtime within the engine
+
+### **General steps to run an inference using the Ort::Api**
+1. Include and link [ONNX Runtime libs](https://github.com/microsoft/onnxruntime)
+2. Create the [Ort::Env](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_env.html)
+3. Specify [SessionOptions](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_session_options.html)
+4. Create [Ort::Session](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_session.html)
+5. Create [Ort::MemoryInfo](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_memory_info.html)
+6. Create [Ort::Allocator](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_allocator_with_default_options.html)
+7. Specify ONNX file location
+8. Specify input names
+9. Specify input shapes
+10. Specify input count
+11. Specify output names
+12. Specify output shapes
+13. Specify output count
+14. Create input vector
+15. Create [input tensor](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_value.html#a3898146b5fc35b838bd48db807dd6c8e) from input vector
+16. Create output vector
+17. Create [output tensor](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_value.html#a3898146b5fc35b838bd48db807dd6c8e) from output vector
+18. Configure [RunOptions](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_run_options.html)
+19. [Run inference](https://onnxruntime.ai/docs/api/c/struct_ort_1_1_session.html#ae1149a662d2a2e218e5740672bbf0ebe)
+20. Retrieve data from output tensor
+
+- These are broadly the steps that the ONNX Gem uses in order to run an inference, however as opposed to the Ort::Api, most of these are hidden from the end user
+
+### **Help me understand the code?**
+The gem is fairly well commented, and should give you a general idea of how it works. I recommend starting with [Model.h](https://github.com/o3de/o3de-extras/blob/onnx-experimental/Gems/ONNX/Code/Include/ONNX/Model.h) as that it where the Model class lives. Along with the general steps to run an inference using the Ort::Api above, as well as the [documentation for the Ort::Api](https://onnxruntime.ai/docs/api/c/namespace_ort.html), take a look at:
```

Comment:
nit: It's usually good practice to avoid 'I' in these kind of docs - 'We' is better, or rephrase to take out the personal element - e.g. It is recommended to start with Model...

Reply:
Changed

Reply:
Thanks!